src/content_sao.cpp

```diff
@@ -990,7 +990,8 @@ void PlayerSAO::getStaticData(std::string * result) const
 
 void PlayerSAO::step(float dtime, bool send_recommended)
 {
-	if (m_drowning_interval.step(dtime, 2.0f)) {
+	bool immortal = itemgroup_get(m_armor_groups, "immortal") == 1;
+	if (m_drowning_interval.step(dtime, 2.0f) && (!immortal)) {
```

Comment:
Please create a new function (for example) `PlayerSAO::isVulnerable` to check against the conditions
1) whether PvP is enabled
2) `immortal` armor group is set